src/wavm.cpp

```diff
@@ -268,21 +275,28 @@ ExecutionResult WavmEngine::internalExecute(
   ensureCondition(mainFunction, ContractValidationFailure, "\"main\" not found");
 
   // this is how WAVM's try/catch for exceptions
-  Runtime::catchRuntimeExceptions(
-    [&] {
-      try {
-        vector<IR::Value> invokeArgs;
-        Runtime::invokeFunctionChecked(wavm_context, mainFunction, invokeArgs);
-      } catch (EndExecution const&) {
-        // This exception is ignored here because we consider it to be a success.
-        // It is only a clutch for POSIX style exit()
+  // TODO: this is terrible, need to clean up the handling of `interface`
+  try {
+    Runtime::catchRuntimeExceptions(
+      [&] {
+        try {
+          vector<IR::Value> invokeArgs;
+          Runtime::invokeFunctionChecked(wavm_context, mainFunction, invokeArgs);
+        } catch (EndExecution const&) {
+          // This exception is ignored here because we consider it to be a success.
+          // It is only a clutch for POSIX style exit()
+        }
+      },
+      [&](Runtime::Exception&& exception) {
+        // FIXME: decide if each of the exception fit into VMTrap/InternalError
+        ensureCondition(false, VMTrap, Runtime::describeException(exception));
       }
-    },
-    [&](Runtime::Exception&& exception) {
-      // FIXME: decide if each of the exception fit into VMTrap/InternalError
-      ensureCondition(false, VMTrap, Runtime::describeException(exception));
-    }
-  );
+    );
+  } catch (exception const& e) {
+    // clean up
+    wavm_host_module::interface.pop();
```

Comment:
The `interface` contains raw pointers, are those deleted?

Reply:
Is this `wavm_host_module` required by WAVM? It's not thread-save to have a global structure handling execution.

Reply:
It is required due to that fact the host functions can be declared via macros only :(

Reply:
> The `interface` contains raw pointers, are those deleted?

That code was I think suggested by you, can you please check?

Reply:
Yes, it is allocated on the stack.